### PR TITLE
Create basic type definitions for swagger-ui

### PR DESCRIPTION
**Pull request by Juarrow**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project. (Started with dts-gen and adjusted/threw out everything i could not use, yet, etc]
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/swagger-ui/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,323 @@
+// Type definitions for swagger-ui 3.47
+// Project: https://github.com/swagger-api/swagger-ui
+// Definitions by: Julian Pfeil <https://github.com/juarrow/>
```
Comment (peterblazejewicz) [inline types/swagger-ui/index.d.ts:3]:
```suggestion
// Definitions by: Julian Pfeil <https://github.com/juarrow>
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Juarrow thx!

**Review by legit100LP — APPROVED** [review]

**Comment by legit100LP** [comment]
How do I apply all of the script into my iPhone ? Also you are welcome and yes most certainly will take another view of the good work

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Juarrow** [comment]
@peterblazejewicz Thx for the approval!

May you tell me what is wrong with my tsconfig? The thing that comes to mind is that i did add `dom` to `lib: []`.

**Comment by Juarrow** [comment]
> How do I apply all of the script into my iPhone ?

I dont' understand your question, to be honest. May you re-phrase it, please?

**Comment by legit100LP** [comment]
Let me start over. I have been trying to figure out how to jailbreak my iphone and I am new to the GitHub so I clearly don’t know how to do stuff but was curious if you would explain to me if you are jailbroken and how you went about doing so ?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Juarrow** [comment]
> Let me start over. I have been trying to figure out how to jailbreak my iphone and I am new to the GitHub so I clearly don’t know how to do stuff but was curious if you would explain to me if you are jailbroken and how you went about doing so ?

@legit100LP i think you are in the wrong GitHub repository for those kind of questions (not sure if GitHub is the right place at all)

**Comment by peterblazejewicz** [comment]
> May you tell me what is wrong with my tsconfig? The thing that comes to mind is that i did add dom to lib: [].

I don't think there is a need for 'dom' dependency in this definition. I think you could just drop 'dom'.
The label `check config` gets added when `tsconfig.json` differs from default one I think.

**Comment by Juarrow** [comment]
Not sure what the default one is, but the one i have in here is the one created automatically by `dts-gen` with the addition of "DOM" in `lib: []`, which -(ironically) according to the link the bot lists- should be allowed.

**Comment by peterblazejewicz** [comment]
you need 'dom' due to `HTMLElement` being used, that's OK

**Review by peterblazejewicz — APPROVED** [review]

**Comment by elibarzilay** [comment]
@peterblazejewicz, @Juarrow: the reason for the "check config" was that the bot was only considering `DOM` as an unsuspectful change.  I made it ignore the case now.

**Pull request merged**